### DISCUSSION TITLE
[Enterprise Metadata] Allowed restrictions disappears when opening the class definition

### DISCUSSION THREAD

**Discussion by Maksonis** [pimcore/pimcore#13111] [Q&A / Support] [answered]
In my class definition I have Many to Many relation field, and I check allow Objects checkbox with allowed classes Car (for example):

![image](https://user-images.githubusercontent.com/17277953/189291444-19c9a17a-cc44-48e2-9071-0fa4167a1087.png)

and save it. Next time when I go back to Class definition all allowed classes are gone. Same for any time of data (assets, documents).

![image](https://user-images.githubusercontent.com/17277953/189291596-c7f3b015-8a58-4411-83ae-3adaf25d008f.png)

Is this a bug ?

**Comment by dvesh3** [comment]
@Maksonis Seems like a bug. have you tried the same on DataObject class definition?

**Reply by Maksonis** [reply]
@dvesh3 yes, on Data Objects is fine. I noticed this only for Enterprise Metadata class definition. And also Subtype is gone after reloading asset:

![image](https://user-images.githubusercontent.com/17277953/189826774-2d149d26-15ee-42bb-b7b4-97cca66947f1.png)

**Reply by dvesh3** [reply]
I created an issue for the enterprise bundle to fix the problem. thanks!

**Comment by dvesh3** [comment, marked as answer]
Fixed in v1.2.7
